restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -1328,6 +1328,12 @@ private void onCallStateChanged(Object message, ActorRef sender) throws Transiti
                     if (dialBranches != null && dialBranches.contains(sender)) {
                         dialBranches.remove(sender);
                     }
+                    if (dialBranches == null || dialBranches.size() == 0){
+                        // https://github.com/RestComm/Restcomm-Connect/issues/2895
+                        // Race condition If there is Play/Say Verb after dial verb, if the Dia verb is failing to call to the other,
+                        // NTFY to stop ringing tone also stop RQNT for playing/say verb. Better to stop it here.
+                        call.tell(new StopMediaGroup(), self());
```

Comment:
is this goign to fix the raceCondition by proper state transition? or just send the stopTone(RQNT) ealier so the chances for race condition are lower?

Reply:
@jaimecasero send the stopTone(RQNT) ealier so the chances for race condition are lower.

Reply:
i thought so, we may need to introduce proper states to ensure consistent behavior. We need to discuss with rest of team.

Reply:
@jaimecasero The ticket has been created for longer than 1 month, Can we accept this fix. I will create jira ticket and The team should fix this later by correct state machine?

Reply:
Jira with logs: https://telestax.atlassian.net/browse/RESTCOMM-2087

Reply:
@jaimecasero The fix just working when enabled-200-ok-delay is set to true.
When enabled-200-ok-delay is set to false. The Fix is not working.

So I think We should fix this by corresponding state transition.